src/components/MAPI/permissions/useUseCasesPermissions.ts

```diff
@@ -38,55 +38,49 @@ function getCombinedPermissions(
 export function useUseCasesPermissions(
   useCases: IPermissionsUseCase[] | null,
   subjectName: string = '',
-  subjectType: SubjectType = SubjectType.Myself
+  subjectType: SubjectTypes = SubjectTypes.Myself
 ): {
   data: IPermissionMap | undefined;
   error: GenericResponseError | undefined;
   isLoading: boolean;
 } {
-  const currentSubject = useMemo(() => {
-    if (subjectName === '') return undefined;
+  const loggedInUser = useSelector(getLoggedInUser);
+  const impersonation = useSelector(
+    (state: IState) => state.main.impersonation
+  );
+
+  const currentSubject: IPermissionsSubject | undefined = useMemo(() => {
+    if (subjectName === '' && subjectType !== SubjectTypes.Myself)
+      return undefined;
 
     switch (subjectType) {
-      case SubjectType.User:
+      case SubjectTypes.Myself:
+        return (
+          impersonation || {
+            user: loggedInUser?.email,
+            groups: loggedInUser?.groups,
+          }
+        );
+      case SubjectTypes.User:
         return { user: subjectName };
-      case SubjectType.Group:
+      case SubjectTypes.Group:
         return { groups: [subjectName] };
+      case SubjectTypes.ServiceAccount:
+        return { serviceAccount: subjectName };
       default:
         return undefined;
     }
-  }, [subjectName, subjectType]);
+  }, [impersonation, loggedInUser, subjectName, subjectType]);
 
+  const useNamespacePermissions =
+    subjectType === SubjectTypes.Myself
+      ? usePermissions
+      : useSubjectPermissions;
```

Comment:
I tried to simplify the logic for using `usePermissions` and `useSubjectPermissions` to get namespace permissions and arrived at this - not sure if it's an anti-pattern.

Reply:
If it works, let's do it like this. It definitely simplifies the logic 👍